apps/rush/src/actions/VersionAction.ts

```diff
@@ -77,82 +87,60 @@ export default class VersionAction extends BaseRushAction {
       }
     }
 
-    this._versionManager = new VersionManager(this.rushConfiguration);
+    this._versionManager = new VersionManager(this.rushConfiguration, this._getUserEmail());
     if (this._ensureVersionPolicy.value) {
       const tempBranch: string = 'version/ensure-' + new Date().getTime();
-      const git: Git = new Git(this._targetBranch.value);
+      this._versionManager.ensure(this._versionPolicy.value, true);
 
-      const updatedPackages: Map<string, IPackageJson> = this._versionManager.ensure(this._versionPolicy.value);
-      if (updatedPackages) {
+      const updatedPackages: Map<string, IPackageJson> = this._versionManager.updatedProjects;
+      if (updatedPackages.size > 0) {
         console.log(`${updatedPackages.size} packages are getting updated.`);
-        this._updateFiles(updatedPackages);
-        // Make changes in temp branch.
-        git.checkout(tempBranch, true);
-
-        // Stage, commit, and push the changes to remote temp branch.
-        git.addChanges();
-        git.commit();
-        git.push(tempBranch);
-
-        // Now merge to target branch.
-        git.checkout(this._targetBranch.value);
-        git.pull();
-        git.merge(tempBranch);
-        git.push(this._targetBranch.value);
-        git.deleteBranch(tempBranch);
+        this._gitProcess(tempBranch);
       }
+    } else if (this._bumpVersion.value) {
+      const tempBranch: string = 'version/bump-' + new Date().getTime();
+      this._versionManager.bump(this._versionPolicy.value,
+        BumpType[this._overwriteBump.value],
+        this._prereleaseIdentifier.value,
+        true);
+      this._gitProcess(tempBranch);
     }
   }
 
-  private _updateFiles(updatedPackages: Map<string, IPackageJson>): void {
-    updatedPackages.forEach((newPackageJson, packageName) => {
-      const rushProject: RushConfigurationProject = this.rushConfiguration.getProjectByName(packageName);
-      // Update package.json
-      const packagePath: string = path.join(rushProject.projectFolder, 'package.json');
-      fsx.writeFileSync(packagePath, JSON.stringify(newPackageJson, undefined, 2), { encoding: 'utf8' });
-
-      if (newPackageJson.version !== rushProject.packageJson.version) {
-        // If package version changes, add an entry to changelog
-        const change: IChangeInfo = this._createChangeInfo(newPackageJson, rushProject);
-        ChangelogGenerator.updateIndividualChangelog(change,
-          rushProject.projectFolder,
-          true);
-      }
-      // TODO: if only package dependency changes, add change file.
-    });
+  private _getUserEmail(): string {
+    return Utilities.executeCommandAndCaptureOutput('git',
+        ['config', 'user.email'], '.').trim();
```

Comment:
agree. This is not the only place. We will do another checkin to sort those out.

---
In reply to: [134068909](https://github.com/Microsoft/web-build-tools/pull/277#discussion_r134068909) [](ancestors = 134068909)